helm/blueapi/templates/ingress.yaml

```diff
@@ -1,22 +1,43 @@
-{{- if .Values.ingress.create -}}
+{{- if .Values.ingress.enabled -}}
 apiVersion: networking.k8s.io/v1
 kind: Ingress
 metadata:
   name: {{ include "blueapi.fullname" . }}
+  labels:
+    {{- include "blueapi.labels" . | nindent 4 }}
+  {{- with .Values.ingress.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
 spec:
-  ingressClassName: nginx
+  {{- with .Values.ingress.className }}
+  ingressClassName: {{ . }}
+  {{- end }}
+  {{- if .Values.ingress.tls }}
   tls:
-  - hosts:
-    - {{ required "A valid hostname must be provided" .Values.ingress.host }}
+    {{- range .Values.ingress.tls }}
+    - hosts:
+        {{- range .hosts }}
+        - {{ . | quote }}
+        {{- end }}
+      secretName: {{ .secretName }}
+    {{- end }}
+  {{- end }}
```

Comment:
Should: Write a test to prove this still works and doesn't change the output, DLS ingress controller is a bit pedantic.

Reply:
Wasn't that the purpose of #871?
Here is a git diff of before/after these changes with the ingress create/enabled. 
Annoyingly, github won't allow attaching yaml files

![image](https://github.com/user-attachments/assets/7d7fcb85-fbf3-4925-8525-22bdc5314b49)

Reply:
#871 wasn't meant to be an exhaustive set of tests, just a framework/starting point. Right now we are confident that our chart works because we've experienced it working in production, but changes are risky. The idea was that people would write new tests as and when they changed/added things to increase confident and reduce the need to test on the cluster over time. In other words: You should write a test for everything this PR touches that doesn't already have a test.

Reply:
This change now adds tests that check the structure of the Ingress (and service) as the first commit, then as a final commit make the changes to the inputs (but not outputs!) that are expected, to prove that the new system outputs the same as the old system.
![image](https://github.com/user-attachments/assets/9d8467cc-0765-4d0f-bd18-35fc3f0661a5)